src/test/java/com/esri/core/geometry/TestOGC.java

```diff
@@ -955,5 +955,60 @@ public void testEmptyBoundary() throws Exception {
 		}
 	}
 	
+	@Test
+	public void testUnionPointWithEmptyLineString() {
+		assertUnion("POINT (1 2)", "LINESTRING EMPTY", "GEOMETRYCOLLECTION (POINT (1 2))");
+	}
+
+	@Test
+	public void testUnionPointWithLinestring() {
+		assertUnion("POINT (1 2)", "LINESTRING (3 4, 5 6)", "GEOMETRYCOLLECTION (POINT (1 2), LINESTRING (3 4, 5 6))");
+	}
+
+	@Test
+	public void testUnionLinestringWithEmptyPolygon() {
+		assertUnion("LINESTRING (1 2, 3 4)", "POLYGON EMPTY", "GEOMETRYCOLLECTION (LINESTRING (1 2, 3 4))");
+	}
+
+	@Test
+	public void testUnionLinestringWithPolygon() {
+		assertUnion("LINESTRING (1 2, 3 4)", "POLYGON ((0 0, 1 1, 0 1, 0 0))",
+				"GEOMETRYCOLLECTION (LINESTRING (1 2, 3 4), POLYGON ((0 0, 1 1, 0 1, 0 0)))");
+	}
 	
+	@Test
+	public void testUnionGeometryCollectionWithGeometryCollection() {
+		assertUnion("GEOMETRYCOLLECTION (LINESTRING (1 2, 3 4), POLYGON ((0 0, 1 1, 0 1, 0 0)))", 
+				"GEOMETRYCOLLECTION (POINT (1 2), POINT (2 3), POINT (0.5 0.5), POINT (3 5), LINESTRING (3 4, 5 6), POLYGON ((0 0, 1 0, 1 1, 0 0)))",
+				"GEOMETRYCOLLECTION (POINT (3 5), LINESTRING (1 2, 2 3, 3 4, 5 6), POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0)))");
+	}
+
+	private void assertUnion(String leftWkt, String rightWkt, String expectedWkt) {
+		OGCGeometry union = OGCGeometry.fromText(leftWkt).union(OGCGeometry.fromText(rightWkt));
+		assertEquals(expectedWkt, union.asText());
+	}
+	
+	@Test
+	public void testDisjointOnGeometryCollection() {
+		OGCGeometry ogcGeometry = OGCGeometry.fromText("GEOMETRYCOLLECTION (POINT (1 1))");
+		assertFalse(ogcGeometry.disjoint(OGCGeometry.fromText("POINT (1 1)")));
+	}
+
+	@Test
+	public void testContainsOnGeometryCollection() {
+		OGCGeometry ogcGeometry = OGCGeometry.fromText("GEOMETRYCOLLECTION (POINT (1 1))");
+		assertTrue(ogcGeometry.contains(OGCGeometry.fromText("POINT (1 1)")));
+	}
+
+	@Test
+	public void testIntersectsOnGeometryCollection() {
```

Comment:
I modified the test to ensure that if `intersects` is symmetrical and the test failed with NPE:

```
	@Test
	public void testIntersectsOnGeometryCollection() {
		assertIntersects("GEOMETRYCOLLECTION (POINT (1 1))", "POINT (1 1)", true);
	}

	private void assertIntersects(String wkt, String otherWkt, boolean intersects)
	{
		assertEquals(OGCGeometry.fromText(wkt).intersects(OGCGeometry.fromText(otherWkt)), intersects);
		assertEquals(OGCGeometry.fromText(otherWkt).intersects(OGCGeometry.fromText(wkt)), intersects);
	}
```

Here is the failure:

```
	at com.esri.core.geometry.RelationalOperations.relate(RelationalOperations.java:45)
	at com.esri.core.geometry.OperatorDisjointLocal.execute(OperatorDisjointLocal.java:31)
	at com.esri.core.geometry.GeometryEngine.disjoint(GeometryEngine.java:376)
	at com.esri.core.geometry.ogc.OGCGeometry.disjoint(OGCGeometry.java:275)
	at com.esri.core.geometry.ogc.OGCGeometry.intersects(OGCGeometry.java:280)
	at com.esri.core.geometry.TestOGC.assertIntersects(TestOGC.java:1007)
	at com.esri.core.geometry.TestOGC.testIntersectsOnGeometryCollection(TestOGC.java:1001)
```

Reply:
ok. I'll fix this.

Reply:
Fixed